File: app/src/main/java/com/hhp227/datemate/data/LoungeRepository.kt
package com.hhp227.datemate.data

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.hhp227.datemate.model.Post
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.MutableStateFlow

class LoungeRepository(
    private val rootRef: DatabaseReference = FirebaseDatabase.getInstance().reference,
    private val postRef: DatabaseReference = rootRef.child("posts")
) {
    fun getPosts(): Flow<List<Post>> {
        val mutableStateFlow = MutableStateFlow<List<Post>>(emptyList())

        postRef.get().addOnCompleteListener { task ->
            if (task.isSuccessful) {
                mutableStateFlow.value = task.result.children.mapNotNull { snapshot ->
                    snapshot.key?.let { snapshot.getValue(Post::class.java)?.apply { key = it } } ?: Post()
                }
            }
        }
        return mutableStateFlow
    }

    fun test() = "헬로우"
}

File: iOS/DateMate/Data/LoungeRepository.swift
//
//  LoungeRepository.swift
//  DateMate
//
//  Created by 홍희표 on 2021/12/09.
//

import Foundation
import FirebaseDatabase
import Combine

class LoungeRepository {
    private let rootRef: DatabaseReference
    
    private var postRef: DatabaseReference
    
    func getPosts() -> AnyPublisher<[Post], Error> {
        return postRef.observer(for: .value).tryMap { result in
            result.children.map { dataSnapshot -> Post in
                if let snapshot = dataSnapshot as? DataSnapshot, let dic = snapshot.value as? [String: Any] {
                    let post = Post.init(
                        id: dic["uid"] as! String,
                        author: dic["author"] as! String,
                        title: dic["title"] as! String,
                        body: dic["body"] as! String,
                        starCount: dic["starCount"] as! Int,
                        stars: [:],
                        key: snapshot.key
                    )
                    return post
                } else {
                    fatalError()
                }
            }
        }.eraseToAnyPublisher()
    }
    
    func test() -> String {
        return "헬로우"
    }
    
    init() {
        self.rootRef = Database.database().reference()
        self.postRef = rootRef.child("posts")
    }
}

